core/src/subgraph/instance_manager.rs

```diff
@@ -460,6 +462,10 @@ where
     let id_for_err = ctx.inputs.deployment.hash.clone();
     let mut first_run = true;
 
+    // Exponential backoff that starts with two minutes and keeps
+    // increasing its timeout exponentially until it reaches 48 hours total, aka two days.
+    let mut backoff = ExponentialBackoff::new(MINUTE * 2, MINUTE * 60 * 24 * 2);
```

Comment:
The intention is that the subgraph self-heals _soon_ after the underlying issue is fixed, even if it takes hours or days for the issue to be fixed. 48h is too long to be a plausible definition of _soon_, I'd make the default maximum maybe 30 minutes.  Either way this maximum should be configurable.

Reply:
Retrying forever is the desirable behavior for the subgraph to be able to self-heal even if it takes days for the underlying issue to be fixed.